kind/pkg/cluster/config/any.go

```diff
@@ -0,0 +1,38 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+/*
+This file contains interfaces and code related to all API versions
+*/
+
+// Any represents any API version of Config
+type Any interface {
+	// Validate should return an error of type `*Errors` if config is invalid
+	Validate() error
+	// ToCurrent should convert a config version to the version in this package
+	ToCurrent() *Config
+	// ApplyDefaults should set unset fields to defaults
+	ApplyDefaults()
+	// Version should return the apiVersion for this config
+	Version() string
+}
+
+// Convert converts a convertable config version to the current Config type
+func Convert(config Any) *Config {
```

Comment:
From what I can tell, this function is not used/is dead code?

Reply:
It was meant to be a helper to be a helper for consumers of the API, but probably isn't particularly useful. I should drop it.